In [81]:
import pandas as pd
import plotly as pl

In [82]:
developed_countries = [
    "Australia",
    "Austria",
    "Belgium",
    "Canada",
    "Denmark",
    "Finland",
    "France",
    "Germany",
    "Iceland",
    "Ireland",
    "Israel",
    "Italy",
    "Japan",
    "Luxembourg",
    "Netherlands",
    "New Zealand",
    "Norway",
    "Portugal",
    "Singapore",
    "South Korea",
    "Spain",
    "Sweden",
    "Switzerland",
    "United Kingdom",
    "United States",
    "Andorra",
    "Cyprus",
    "Czech Republic",
    "Estonia",
    "Greece",
    "Hong Kong SAR",
    "Latvia",
    "Lithuania",
    "Malta",
    "Monaco",
    "San Marino",
    "Slovakia",
    "Slovenia"
]

In [83]:
df = pd.read_csv("merged_to_csv")
countries = pd.read_csv('../sus.csv')['Entity'].drop_duplicates()
world_data = df[(df['Element'] == "Emissions (CO2)") & (df['Item'] == "All sectors without LULUCF") & (df['Area'].isin(countries))].drop(columns=['Item', 'Element', 'Unit', 'Entity', 'Access to electricity (% of population)', 'Access to clean fuels for cooking', 'Renewable-electricity-generating-capacity-per-capita', 'Financial flows to developing countries (US $)', 'Financial flows to developing countries (US $)', 'Electricity from fossil fuels (TWh)', 'Electricity from nuclear (TWh)', 'Electricity from renewables (TWh)','Low-carbon electricity (% electricity)', 'Primary energy consumption per capita (kWh/person)', 'Energy intensity level of primary energy (MJ/$2017 PPP GDP)', 'Value_co2_emissions_kt_by_country', 'Renewables (% equivalent primary energy)','gdp_growth', 'gdp_per_capita', 'Density\\n(P/Km2)','Land Area(Km2)', 'Latitude', 'Longitude']).dropna()
world_data

Area  Year   Emission  \
220              Afghanistan  2000     914.00   
693                  Algeria  2000   92071.29   
1141                  Angola  2000   16725.00   
1506     Antigua and Barbuda  2000     295.52   
1740               Argentina  2000  145728.95   
...                      ...   ...        ...   
1165631              Vanuatu  2019     164.34   
1166556                Yemen  2019   17423.00   
1166967               Zambia  2019    6720.00   
1167194             Zimbabwe  2019   11151.00   
1225959             Zimbabwe  2020   10054.00   

         Renewable energy share in the total final energy consumption (%)  
220                                                  44.99                 
693                                                   0.43                 
1141                                                 73.44                 
1506                                                  0.00                 
1740                                                  9.85                 
...                                                    ...                 
1165631                                              31.86                 
1166556                                               3.11                 
1166967                                              84.50                 
1167194                                              81.50                 
1225959                                              81.90                 

[3302 rows x 4 columns]

In [84]:
means = pd.concat([world_data['Renewable energy share in the total final energy consumption (%)'].groupby(world_data['Year']).mean(), world_data['Emission'].groupby(world_data['Year']).sum()], axis=1)
means


Renewable energy share in the total final energy consumption (%)  \
Year                                                                     
2000                                          35.556012                  
2001                                          35.084049                  
2002                                          35.015767                  
2003                                          34.636994                  
2004                                          34.456012                  
2005                                          34.400123                  
2006                                          34.107301                  
2007                                          33.502606                  
2008                                          33.587636                  
2009                                          33.691273                  
2010                                          33.489818                  
2011                                          33.138182                  
2012                                          33.231446                  
2013                                          33.426048                  
2014                                          33.211677                  
2015                                          32.912156                  
2016                                          32.687545                  
2017                                          32.517365                  
2018                                          32.569162                  
2019                                          32.333952                  
2020                                          81.900000                  

          Emission  
Year                
2000  1.545648e+07  
2001  1.564051e+07  
2002  1.609773e+07  
2003  1.727482e+07  
2004  1.838153e+07  
2005  1.922812e+07  
2006  2.018940e+07  
2007  2.098711e+07  
2008  2.186008e+07  
2009  2.199461e+07  
2010  2.318581e+07  
2011  2.432786e+07  
2012  2.502619e+07  
2013  2.520385e+07  
2014  2.547133e+07  
2015  2.558173e+07  
2016  2.559176e+07  
2017  2.600463e+07  
2018  2.652561e+07  
2019  2.673910e+07  
2020  1.005400e+04

In [85]:
"""import plotly.express as px
import plotly.graph_objects as go
"""
import plotly.graph_objects as go
from plotly.subplots import make_subplots

# Create figure with secondary y-axis
fig = make_subplots(specs=[[{"secondary_y": True}]])

# Add traces
fig.add_trace(
    go.Scatter(x=means.index, y=means['Emission'], name="Emissions"),
    secondary_y=False,
)

fig.add_trace(
    go.Scatter(x=means.index, y=means['Renewable energy share in the total final energy consumption (%)'], name="Renewable energy share (%)"),
    secondary_y=True,
)

# Add figure title
fig.update_layout(
    title_text="World average"
)

# Set x-axis title
fig.update_xaxes(title_text="Year")

# Set y-axes titles
fig.update_yaxes(title_text="<b>Emissions</b>", secondary_y=False)
fig.update_yaxes(title_text="<b>Renewable energy share (%)</b>", secondary_y=True)

fig.show()

In [86]:
developed_data = df[(df['Element'] == "Emissions (CO2)") & (df['Item'] == "All sectors without LULUCF") & (df['Area'].isin(developed_countries))].drop(columns=['Item', 'Element', 'Unit','Entity', 'Access to electricity (% of population)','Access to clean fuels for cooking','Renewable-electricity-generating-capacity-per-capita','Financial flows to developing countries (US $)','Electricity from fossil fuels (TWh)', 'Electricity from nuclear (TWh)','Electricity from renewables (TWh)','Low-carbon electricity (% electricity)','Primary energy consumption per capita (kWh/person)','Energy intensity level of primary energy (MJ/$2017 PPP GDP)','Value_co2_emissions_kt_by_country','Renewables (% equivalent primary energy)', 'gdp_growth','gdp_per_capita', 'Density\\n(P/Km2)', 'Land Area(Km2)', 'Latitude','Longitude'])

developed_means = pd.concat([developed_data['Renewable energy share in the total final energy consumption (%)'].groupby(developed_data['Year']).mean(), developed_data['Emission'].groupby(developed_data['Year']).sum()], axis=1)
    

In [87]:
import plotly.graph_objects as go
from plotly.subplots import make_subplots

# Create figure with secondary y-axis
fig = make_subplots(specs=[[{"secondary_y": True}]])

# Add traces
fig.add_trace(
    go.Scatter(x=developed_means.index, y=developed_means['Emission'], name="Developed emissions"),
    secondary_y=False,
)

fig.add_trace(
    go.Scatter(x=developed_means.index, y=developed_means['Renewable energy share in the total final energy consumption (%)'], name="Developed renewable energy share (%)"),
    secondary_y=True,
)

# Add figure title
fig.update_layout(
    title_text="Developed world average"
)

# Set x-axis title
fig.update_xaxes(title_text="Year")

# Set y-axes titles
fig.update_yaxes(title_text="<b>Emissions</b>", secondary_y=False)
fig.update_yaxes(title_text="<b>Renewable energy share (%)</b>", secondary_y=True)

fig.show()

In [88]:
developing_data = df[(df['Element'] == "Emissions (CO2)") & (df['Item'] == "All sectors without LULUCF") & (~df['Area'].isin(developed_countries))].drop(columns=['Item', 'Element', 'Unit','Entity', 'Access to electricity (% of population)','Access to clean fuels for cooking','Renewable-electricity-generating-capacity-per-capita','Financial flows to developing countries (US $)','Electricity from fossil fuels (TWh)', 'Electricity from nuclear (TWh)','Electricity from renewables (TWh)','Low-carbon electricity (% electricity)','Primary energy consumption per capita (kWh/person)','Energy intensity level of primary energy (MJ/$2017 PPP GDP)','Value_co2_emissions_kt_by_country','Renewables (% equivalent primary energy)', 'gdp_growth','gdp_per_capita', 'Density\\n(P/Km2)', 'Land Area(Km2)', 'Latitude','Longitude']).dropna()
developing_means = pd.concat([developing_data['Renewable energy share in the total final energy consumption (%)'].groupby(developing_data['Year']).mean(), developing_data['Emission'].groupby(developing_data['Year']).sum()], axis=1)

In [89]:
import plotly.graph_objects as go
from plotly.subplots import make_subplots

# Create figure with secondary y-axis
fig = make_subplots(specs=[[{"secondary_y": True}]])

# Add traces
fig.add_trace(
    go.Scatter(x=developing_means.index, y=developing_means['Emission'], name="Developing emissions"),
    secondary_y=False,
)

fig.add_trace(
    go.Scatter(x=developing_means.index, y=developing_means['Renewable energy share in the total final energy consumption (%)'], name="Developing renewable energy share (%)"),
    secondary_y=True,
)

# Add figure title
fig.update_layout(
    title_text="Developing world average"
)

# Set x-axis title
fig.update_xaxes(title_text="Year")

# Set y-axes titles
fig.update_yaxes(title_text="<b>Emissions</b>", secondary_y=False)
fig.update_yaxes(title_text="<b>Renewable energy share (%)</b>", secondary_y=True)

fig.show()

In [90]:
china_data = df[(df['Element'] == "Emissions (CO2)") & (df['Item'] == "All sectors without LULUCF") & (df['Area'] == "China")].drop(columns=[ 'Item', 'Element', 'Unit',
    'Entity', 'Access to electricity (% of population)',
    'Access to clean fuels for cooking',
    'Renewable-electricity-generating-capacity-per-capita',
    'Financial flows to developing countries (US $)',
    'Electricity from fossil fuels (TWh)', 'Electricity from nuclear (TWh)',
    'Electricity from renewables (TWh)',
    'Low-carbon electricity (% electricity)',
    'Primary energy consumption per capita (kWh/person)',
    'Energy intensity level of primary energy (MJ/$2017 PPP GDP)',
    'Value_co2_emissions_kt_by_country',
    'Renewables (% equivalent primary energy)', 'gdp_growth',
    'gdp_per_capita', 'Density\\n(P/Km2)', 'Land Area(Km2)', 'Latitude',
    'Longitude']).dropna()
china_means = pd.concat([china_data['Renewable energy share in the total final energy consumption (%)'].groupby(china_data['Year']).mean(), china_data['Emission'].groupby(china_data['Year']).sum()], axis=1)
china_means

Renewable energy share in the total final energy consumption (%)  \
Year                                                                     
2000                                              29.63                  
2001                                              28.36                  
2002                                              27.00                  
2003                                              23.86                  
2004                                              20.17                  
2005                                              17.44                  
2006                                              16.39                  
2007                                              14.88                  
2008                                              14.14                  
2009                                              13.43                  
2010                                              12.26                  
2011                                              11.34                  
2012                                              11.53                  
2013                                              11.45                  
2014                                              11.94                  
2015                                              12.18                  
2016                                              12.56                  
2017                                              13.13                  
2018                                              13.71                  
2019                                              14.45                  

          Emission  
Year                
2000  3.839650e+06  
2001  3.913106e+06  
2002  4.294964e+06  
2003  5.022124e+06  
2004  5.728722e+06  
2005  6.413384e+06  
2006  7.069472e+06  
2007  7.615932e+06  
2008  8.123134e+06  
2009  8.526522e+06  
2010  9.311518e+06  
2011  1.025155e+07  
2012  1.051862e+07  
2013  1.073465e+07  
2014  1.082199e+07  
2015  1.072279e+07  
2016  1.060168e+07  
2017  1.082771e+07  
2018  1.109717e+07  
2019  1.137014e+07

In [91]:
import plotly.graph_objects as go
from plotly.subplots import make_subplots

# Create figure with secondary y-axis
fig = make_subplots(specs=[[{"secondary_y": True}]])

# Add traces
fig.add_trace(
    go.Scatter(x=china_means.index, y=china_means['Emission'], name="China emissions"),
    secondary_y=False,
)

fig.add_trace(
    go.Scatter(x=china_means.index, y=china_means['Renewable energy share in the total final energy consumption (%)'], name="China renewable energy share (%)"),
    secondary_y=True,
)

# Add figure title
fig.update_layout(
    title_text="China world average"
)

# Set x-axis title
fig.update_xaxes(title_text="Year")

# Set y-axes titles
fig.update_yaxes(title_text="<b>China emissions</b>", secondary_y=False)
fig.update_yaxes(title_text="<b>China renewable energy share (%)</b>", secondary_y=True)

fig.show()

The world becoming more sustainable is not being substantiated according to the graphs. When looking at the world sum of emissions, it has grown since 2000 till 2019 by almost 40%. Even though the emissions from the developed world have reduced it’s emissions by almost 1.1 million, China has increased it’s emissions by almost 7 million, in the same time period. This increase of emissions done by China offsets the reductions of emissions made by developed countries largely. 

With our best efforts of becoming sustainable, it hasn’t resulted in any improvements worldwide. It turns out that world pollution is mostly driven by developing countries. Developing countries produce twice as much emissions as the developed world. Therefore even if we reduce the emissions of developed countries to zero, developing countries would still be responsible for 60% of global emissions.  A different solution to world pollution could be a worldwide carbon tax. This would ensure that all countries contribute to the reduction of emissions. 
